webviz_subsurface/plugins/_tornado_plotter_fmu/_plugin.py

```diff
@@ -0,0 +1,277 @@
+
+        provider_factory = EnsembleTableProviderFactory.instance()
+        self.error_message = ""
+
+        # Reading from the csv file
+        if ensemble is not None and csvfile is not None:
+            ens_path = webviz_settings.shared_settings["scratch_ensembles"][ensemble]
+            self._parameter_provider = (
+                provider_factory.create_from_per_realization_parameter_file(ens_path)
+            )
+            self._table_provider = (
+                provider_factory.create_from_per_realization_csv_file(ens_path, csvfile)
+            )
+            self._ensemble_name = ensemble
+
+        elif aggregated_csvfile and aggregated_parameterfile is not None:
+            self._parameter_provider = provider_factory.create_from_ensemble_csv_file(
+                aggregated_parameterfile
+            )
+            self._table_provider = provider_factory.create_from_ensemble_csv_file(
+                aggregated_csvfile
+            )
+            try:
+                self._ensemble_name = self._table_provider.get_column_data(
+                    column_names=["ENSEMBLE"]
+                )
+            except KeyError:
+                self.error_message = f"'ENSEMBLE' is missing from {aggregated_csvfile}"
+
+        else:
+            self.error_message = (
+                "Specify either ensembles and csvfile or aggregated_csvfile "
+                "and aggregated_parameterfile"
+            )
+
+        try:
+            design_matrix_df = self._parameter_provider.get_column_data(
+                column_names=["SENSNAME", "SENSCASE"]
+            )
+        except KeyError:
+            self.error_message = (
+                "Required columns 'SENSNAME' and 'SENSCASE' is missing "
+                f"from {self._ensemble_name}. Cannot calculate tornado plots"
+            )
+
+        # Defining the datafame
+        design_matrix_df["ENSEMBLE"] = self._ensemble_name
+        design_matrix_df["SENSTYPE"] = design_matrix_df.apply(
+            lambda row: find_sens_type(row.SENSCASE), axis=1
+        )
+
+        self._tornado_widget = TornadoWidget(
+            webviz_settings=webviz_settings, realizations=design_matrix_df
+        )
+        self._responses: List[str] = self._table_provider.column_names()
+        if self._single_filters:
+            self._responses = [
+                response
+                for response in self._responses
+                if response not in self._single_filters
+            ]
+        if self._multi_filters:
+            self._responses = [
+                response
+                for response in self._responses
+                if response not in self._multi_filters
+            ]
+        self._initial_response: str = (
+            initial_response if initial_response else self._responses[0]
+        )
+
+        # Settingsgroup for switching between table and bars
+        self.add_store(
```

Comment:
A general comment on use of dcc.Store:
These will add an additional roundtrip between server/client which will reduce performance.
A suggestion is to reduce the amount of dcc.Store calls to increase the performance of the application, especially in cases where it is only used to forward unchanged data between two callbacks.

Reply:
Hi! I removed most stores as you suggested, but the plugin does not work anymore. I get complaints that the callbacks has input from non existing components, because the inputs are directly from the component and not from a store. These complaints are removed when I add the store again. Should I then add the stores, or is there another way of solving this issue?

Reply:
@vvahlin just a wildshot from my side who hasn't read the code, but did you change the inputs of the callbacks to use the ids and properties of the components directly when removing the stores (so instead of using something like Input(selector_store_id, "data") you e.g. use Input(selector_id, "value") )? Cause if you kept Input(selector_store_id, "data") it would make sense that it was a non-existing component.

Reply:
I eventually remembered to change, and still got the complaint:/

Reply:
ok, good to know :) could be something related to the order of components being initialized. We have faced similar issues before. Think also @EirikPeirik had similar issues in another plugin. If you have time, maybe you can ask him if he figured it out?